### ISSUE TITLE
Following some custom domains leads to subscribing to the underlying wp.com subdomain

### ISSUE THREAD

**Issue by rachelmcr** [Automattic/wp-telegram-bot#22]
I confirmed that I could follow and unfollow WordPress.com Simple sites with custom domains and Jetpack sites as expected.

However, following a WordPress.com Atomic site (e.g. `rachelmcr-test-domain.blog`) or a domain linked to a WordPress.com site through a domain mapping or site redirect record that isn't actually a WordPress site (e.g. `google.com`) leads to unexpected results:

1. Enter the command for an Atomic site, such as `follow https://rachelmcr-test-domain.blog`.
2. The bot responds with the message `Now following rachelmcrtestdomain.wordpress.com` (the underlying subdomain).
3. Publish a post on the site.
4. The bot shares the post in the channel with the correct custom domain.
5. Try to unfollow the site with `unfollow https://rachelmcr-test-domain.blog`.
6. The bot responds with the error `It seems you were not following https://rachelmcr-test-domain.blog`.
7. Try to unfollow the site with the subdomain: `unfollow https://rachelmcrtestdomain.wordpress.com`.
8. The bot responds with the message `No longer following https://rachelmcrtestdomain.wordpress.com`.

This is confusing but also makes it hard to unfollow a blog like this if any time has passed since following it, because the underlying subdomain isn't discoverable.

**Comment by markryall** [comment]
Yes - this seems to be an issue with the XMPP functionality and seems specific to atomic sites (I haven't seen it for any other kind of jetpack site).

It still would be possible to subscribe to a WordPress.com site using something other than the primary domain and you would get this kind of behaviour but i'm not sure why the primary domain isn't being used in the case of atomic sites.
